app/controllers/SmallProducerDetailsController.scala

```diff
@@ -18,57 +18,56 @@ package controllers
-                                         navigator: Navigator,
-                                         identify: IdentifierAction,
-                                         getData: DataRetrievalAction,
-                                         requireData: DataRequiredAction,
-                                         formProvider: SmallProducerDetailsFormProvider,
-                                         val controllerComponents: MessagesControllerComponents,
-                                         view: SmallProducerDetailsView
-                                 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+                                                override val messagesApi: MessagesApi,
+                                                sessionRepository: SessionRepository,
+                                                navigator: Navigator,
+                                                identify: IdentifierAction,
+                                                getData: DataRetrievalAction,
+                                                requireData: DataRequiredAction,
+                                                formProvider: SmallProducerDetailsFormProvider,
+                                                val controllerComponents: MessagesControllerComponents,
+                                                view: SmallProducerDetailsView
+                                              )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
   val form = formProvider()
 
   val spList = List(SmallProducer("ABC Ltd", "SDIL123456", (1000L, 1000L)),
     SmallProducer("XYZ Ltd", "SDIL123789", (1000L, 1000L)))
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) {
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
+
+      val preparedForm = request.userAnswers.get(SmallProducerDetailsPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
       val smallProducersSummaryList: List[SummaryListRow] = SmallProducerDetailsSummary.row2(spList)
       val list: SummaryList = SummaryListViewModel(
         rows = smallProducersSummaryList
       )
-      val preparedForm = request.userAnswers.flatMap(_.get(SmallProducerDetailsPage)) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
 
       Ok(view(preparedForm, mode, list))
   }
 
-  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData).async {
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-    val answers = request.userAnswers.getOrElse(UserAnswers(id = request.sdilEnrolment))
       val smallProducersSummaryList: List[SummaryListRow] = SmallProducerDetailsSummary.row2(spList)
```

Comment:
is it possible to  do this calculation one time and not again as done at line no 61?